### PR TITLE
Add Admin::CanonicalEmailBlocks methods

### LINKED ISSUE #331: Implement Admin Canonical Email Blocks API
Mastodon's Admin Canonical Email Blocks API allows an admin to block certain email addresses by their hash. The goal of this task is to implement the following endpoints of this API in BigBone:

- List all canonical email blocks
- Show a single canonical email block
- Test
- Block a canonical email
- Delete a canonical email block

The Admin Canonical Email Blocks API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/admin/canonical_email_blocks/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

Adds methods to get all or one, delete, and add canonical email block(s). Also adds the “test” method which I’ve renamed a more expressive `canonicalizeAndHashEmailAddress`.

Closes #331.

# Type of Change

- New feature
- Documentation

# Breaking Changes

- None

# How Has This Been Tested?

I’ve added new unit tests for all methods and variants.

# Mandatory Checklist

- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [x] I have added KDoc documentation to all public methods

# Optional Things To Check

The items below are some more things to check before asking other people to review your code.

- [x] In case you worked on a new feature: Did you also implement the reactive endpoint (bigbone-rx)?
- [x] In case you added new `*Methods` classes: Did you also reference it in the `MastodonClient` main class?
- [x] Did you also update the documentation in the `/docs` folder (e.g. API Coverage page)?

### CODE REVIEW

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminCanonicalEmailBlockMethods.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    /**
+     * Show a single canonical email block.
+     *
+     * @param id The ID of the Admin::CanonicalEmailBlock in the database.
+     *
+     * @see <a href="https://docs.joinmastodon.org/methods/admin/canonical_email_blocks/#get-one">
+     *     Mastodon API documentation: admin/canonical_email_blocks/#get-one</a>
+     */
+    fun getEmailDomainBlock(id: String): Single<AdminCanonicalEmailBlock> = Single.fromCallable {
```
Comment (andregasser) [inline bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminCanonicalEmailBlockMethods.kt:43]: I think this method should be called `getCanonicalEmailBlock(...)`.

```suggestion
    fun getCanonicalEmailBlock(id: String): Single<AdminCanonicalEmailBlock> = Single.fromCallable {
```

Comment (PattaFeuFeu) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminCanonicalEmailBlockMethods.kt:43]: Whoopsie, Copy-paste mistake. 🙈

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminCanonicalEmailBlockMethods.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    /**
+     * Block a canonical email.
+     *
+     * @param variant One of [BlockCanonicalEmailVariant]s to define how to block a canonical email.
+     *
+     * @see <a href="https://docs.joinmastodon.org/methods/admin/canonical_email_blocks/#create">
+     *     Mastodon API documentation: admin/canonical_email_blocks/#create</a>
+     */
+    fun blockEmailDomain(variant: BlockCanonicalEmailVariant): Single<AdminCanonicalEmailBlock> =
```
Comment (andregasser) [inline bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminCanonicalEmailBlockMethods.kt:55]: Here I would also suggest a change in method name:

```suggestion
    fun blockCanonicalEmail(variant: BlockCanonicalEmailVariant): Single<AdminCanonicalEmailBlock> =
```

File: bigbone/src/main/kotlin/social/bigbone/api/method/admin/AdminCanonicalEmailBlockMethods.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    /**
+     * Canonicalize and hash an email address.
+     *
+     * @param emailAddress The email to canonicalize and hash.
+     * @return All matching canonical email blocks.
+     *
+     * @see <a href="https://docs.joinmastodon.org/methods/admin/canonical_email_blocks/#test">
+     *     Mastodon API documentation admin/canonical_email_blocks/#test</a>
+     */
+    fun canonicalizeAndHashEmailAddress(emailAddress: String): MastodonRequest<List<AdminCanonicalEmailBlock>> {
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/method/admin/AdminCanonicalEmailBlockMethods.kt:92]: This name is much better than `test` - great improvement 👍

**Review by andregasser — CHANGES_REQUESTED** [review]
This looks very good overall - just found two minor naming issues. Good job!

**Comment by PattaFeuFeu** [comment]
@andregasser Done! Thanks for your review. 😊

**Review by andregasser — APPROVED** [review]
👍

**Pull request merged**